hoot-core/src/main/cpp/hoot/core/elements/Node.cpp

```diff
@@ -124,7 +124,7 @@ void Node::visitRo(const ElementProvider& map, ConstElementVisitor& filter) cons
 
 void Node::visitRw(ElementProvider& map, ConstElementVisitor& filter)
 {
-  filter.visit(map.getNode(getId()));
+  filter.visit(boost::dynamic_pointer_cast<const Node>(map.getNode(getId())));
 }
 
```

Comment:
With the inheritance added to ElementVisitor, this cast to const must now be explicitly made.  Several of these throughout.